kaggle_environments/agent.py

```diff
@@ -55,25 +55,28 @@ def get_last_callable(raw, fallback=None):
         raise InvalidArgument("Invalid raw Python: " + repr(e))
 
 
-def build_agent(raw, environment):
-    if raw in environment.agents:
-        return environment.agents[raw]
+def build_agent(raw, builtin_agents, environment_name):
```

Comment:
The entire environment is not picklable so now we just store the parts we need on the Agent class and pass them in here rather than the entire environment.